### PR TITLE
AKU-768: Create upload display item

### PR DESCRIPTION
**Pull request by MartinDoyleUK**
This addresses issue [AKU-768](https://issues.alfresco.com/jira/browse/AKU-768) which is part of the new upload panel epic. Specifically, this creates a new `_UploadsDisplayMixin` interface from the existing `AlfUploadDisplay` control, and then implements it in a new `UploadMonitor` control. Tests have been updated accordingly.

__No unexpected failures ... ready to merge__

### CODE REVIEW

File: aikau/src/main/resources/alfresco/upload/UploadMonitor.js [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+ */
+
+/**
+ * This module provides display handling for file uploads. It's normally only used by the
+ * [UploadService]{@link module:alfresco/services/UploadService}.
+ *
+ * @module alfresco/upload/UploadMonitor
+ * @extends alfresco/upload/_UploadsDisplayMixin
+ * @author Martin Doyle
+ */
```
Comment (draperd) [inline aikau/src/main/resources/alfresco/upload/UploadMonitor.js]: Needs a @since tag

File: aikau/src/main/resources/alfresco/upload/_UploadsDisplayMixin.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+/**
+ * This mixin acts as an interface for widgets displaying uploads' progress.
+ *
+ * @module alfresco/upload/_UploadsDisplayMixin
+ * @extends external:dijit/_WidgetBase
+ * @mixes external:dojo/_TemplatedMixin
+ * @mixes module:alfresco/core/Core
+ * @author Martin Doyle
+ */
```
Comment (draperd) [inline aikau/src/main/resources/alfresco/upload/_UploadsDisplayMixin.js]: Needs @since tag

File: aikau/src/main/resources/alfresco/util/formatUtils.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Utility object for formatting-related utilities. Note that this is not a Class, and so does
```
Comment (draperd) [inline aikau/src/main/resources/alfresco/util/formatUtils.js]: If this is going to be replacing the alfresco/core/FileSizeMixin module then we probably want to deprecate that module. We should probably also reference it in this JSDoc statement.

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/UploadMockXhr.js [hunk 4]
```diff
@@ -43,23 +45,47 @@ define(["dojo/_base/declare",
...
+                  request.respond(this.responseCode, headers, body);
+               }, i * 20);
+
+            }));
+         } catch (e) {
             this.alfLog("error", "The following error occurred setting up the mock server", e);
          }
+      },
+
+      sendProgress: function(request, percent, delay) {
```
Comment (draperd) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/UploadMockXhr.js]: I know it won't appear in the JSDoc output (at least not yet, although we did move MockXhr)... it would be nice to have the JSDoc statement here for consistency.

File: aikau/src/main/resources/alfresco/upload/templates/UploadMonitor.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<div class="${baseClass}">
+   <table class="${baseClass}__items">
+      <thead>
+         <tr class="${baseClass}__item-heading">
+            <td class="${baseClass}__item__name">&nbsp;</td>
+            <td class="${baseClass}__item__progress">&nbsp;</td>
+            <td class="${baseClass}__item__status">&nbsp;</td>
+            <!-- <td class="${baseClass}__item__close-button">&nbsp;</td> -->
```
Comment (draperd) [inline aikau/src/main/resources/alfresco/upload/templates/UploadMonitor.html]: This has presumably been left commented out for later addition in future sprint?

**Comment by draperd** [comment]
Have you missed any files or changes from your commit? I've checked out the remote branch, built and re-started the server but I'm not seeing it working and the unit tests fail.

File: aikau/src/main/resources/alfresco/upload/UploadMonitor.js (hunk 1 above)
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/upload/UploadMonitor.js]: Sorry .... it was late.

File: aikau/src/main/resources/alfresco/upload/_UploadsDisplayMixin.js (hunk 2 above)
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/upload/_UploadsDisplayMixin.js]: As above

File: aikau/src/test/resources/testApp/js/aikau/testing/mockservices/UploadMockXhr.js (hunk 4 above)
Comment (MartinDoyleUK) [inline aikau/src/test/resources/testApp/js/aikau/testing/mockservices/UploadMockXhr.js]: An oversight!

File: aikau/src/main/resources/alfresco/upload/templates/UploadMonitor.html (hunk 5 above)
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/upload/templates/UploadMonitor.html]: Exactly that

**Comment by MartinDoyleUK** [comment]
I don't believe so. Let me try again from a clean restart.

File: aikau/src/main/resources/alfresco/util/formatUtils.js (hunk 3 above)
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/util/formatUtils.js]: Was unaware of this. We should probably discuss this change.
